public/app/core/components/NestedFolderPicker/api.ts

```diff
@@ -0,0 +1,216 @@
+import { createSelector } from '@reduxjs/toolkit';
+import { QueryDefinition, BaseQueryFn } from '@reduxjs/toolkit/dist/query';
+import { QueryActionCreatorResult } from '@reduxjs/toolkit/dist/query/core/buildInitiate';
+import { RequestOptions } from 'http';
+import { useCallback, useEffect, useMemo, useRef } from 'react';
+
+import { ListFolderQueryArgs, browseDashboardsAPI } from 'app/features/browse-dashboards/api/browseDashboardsAPI';
+import { PAGE_SIZE } from 'app/features/browse-dashboards/api/services';
+import { getPaginationPlaceholders } from 'app/features/browse-dashboards/state/utils';
+import { DashboardViewItemWithUIItems, DashboardsTreeItem } from 'app/features/browse-dashboards/types';
+import { RootState } from 'app/store/configureStore';
+import { FolderListItemDTO } from 'app/types';
+import { useDispatch, useSelector } from 'app/types/store';
+
+type ListFoldersQuery = ReturnType<ReturnType<typeof browseDashboardsAPI.endpoints.listFolders.select>>;
+type ListFoldersRequest = QueryActionCreatorResult<
+  QueryDefinition<
+    ListFolderQueryArgs,
+    BaseQueryFn<RequestOptions>,
+    'getFolder',
+    FolderListItemDTO[],
+    'browseDashboardsAPI'
+  >
+>;
+
+const createListFoldersSelector = createSelector(
+  [
+    (parentUid: ListFolderQueryArgs['parentUid']) => parentUid,
+    (parentUid: ListFolderQueryArgs['parentUid'], page: ListFolderQueryArgs['page']) => page,
+    (
+      parentUid: ListFolderQueryArgs['parentUid'],
+      page: ListFolderQueryArgs['page'],
+      limit: ListFolderQueryArgs['limit']
+    ) => limit,
+  ],
+  (parentUid, page, limit) => {
+    return browseDashboardsAPI.endpoints.listFolders.select({ parentUid, page, limit });
+  }
+);
```

Comment:
do we need this? 🤔 feels like we can just `return browseDashboardsAPI.endpoints.listFolders.select({ parentUid, page, limit });` on line 48 directly

Reply:
ahh hmm, good catch. I'll be honest, I have no idea what's going on here - getting the selector right was actually probably the hardest part of the PR 😅

I'm happy to give this a try, but if it doesn't immediately work I'll ask you for working code if you want it changed 💜

Reply:
This worked. Thanks for the fresh eyes Ash!! 👏